airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/SshTunnel.java

```diff
@@ -0,0 +1,245 @@
+      if (tunnelSession != null) {
+        tunnelSession.close();
+        tunnelSession = null;
+      }
+      if (sshclient != null) {
+        sshclient.stop();
+        sshclient = null;
+      }
+    } catch (final Throwable t) {
+      throw new RuntimeException(t);
+    }
+  }
+
+  /**
+   * From the RSA format private key string, use bouncycastle to deserialize the key pair, reconstruct
+   * the keys from the key info, and return the key pair for use in authentication.
+   */
+  private KeyPair getPrivateKeyPair() throws IOException {
+    final PEMParser pemParser = new PEMParser(new StringReader(sshkey));
+    final PEMKeyPair keypair = (PEMKeyPair) pemParser.readObject();
+    final JcaPEMKeyConverter converter = new JcaPEMKeyConverter();
+    return new KeyPair(
+        converter.getPublicKey(SubjectPublicKeyInfo.getInstance(keypair.getPublicKeyInfo())),
+        converter.getPrivateKey(keypair.getPrivateKeyInfo()));
+  }
+
+  /**
+   * Generates a new ssh client and returns it, with forwarding set to accept all types; use this
+   * before opening a tunnel.
+   */
+  private SshClient createClient() {
+    java.security.Security.addProvider(
+        new org.bouncycastle.jce.provider.BouncyCastleProvider());
+    final SshClient client = SshClient.setUpDefaultClient();
+    client.setForwardingFilter(AcceptAllForwardingFilter.INSTANCE);
+    client.setServerKeyVerifier(AcceptAllServerKeyVerifier.INSTANCE);
+    return client;
+  }
+
+  /**
+   * Starts an ssh session; wrap this in a try-finally and use closeTunnel() to close it.
+   */
+  private ClientSession openTunnel(final SshClient client) {
+    try {
+      client.start();
+      final ClientSession session = client.connect(
+          user.trim(),
+          host.trim(),
+          Integer.parseInt(tunnelSshPort.trim())).verify(TIMEOUT_MILLIS)
+          .getSession();
+      if (method.equals(Method.SSH_KEY_AUTH)) {
+        session.addPublicKeyIdentity(getPrivateKeyPair());
+      }
+      if (method.equals(Method.SSH_PASSWORD_AUTH)) {
+        session.addPasswordIdentity(password);
+      }
+      session.auth().verify(TIMEOUT_MILLIS);
+      final SshdSocketAddress address = session.startLocalPortForwarding(
+          new SshdSocketAddress(SshdSocketAddress.LOCALHOST_ADDRESS.getHostName(), Integer.parseInt(tunnelDatabasePort.trim())),
+          new SshdSocketAddress(remoteDatabaseHost.trim(), Integer.parseInt(remoteDatabasePort.trim())));
+      LOGGER.info("Established tunneling session.  Port forwarding started on " + address.toInetSocketAddress());
+      return session;
+    } catch (final IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public String toString() {
+    return "SSHTunnel{" +
+        "method=" + method +
+        ", host='" + host + '\'' +
+        ", tunnelSshPort='" + tunnelSshPort + '\'' +
+        ", user='" + user + '\'' +
+        ", sshkey='" + sshkey + '\'' +
+        ", password='" + password + '\'' +
```

Comment:
The secrets (key, pass) should not be included in the toString method, as it can leak secrets to logs.